code/Test_definitions/device-roaming-status-subscriptions.feature

```diff
@@ -0,0 +1,238 @@
+    And a valid subscription request body  
+    And the request body property "$.type" is "roaming-change-country"	
+    When the request "createSubscription" is sent
+    Then the response code is 201
+    And if the device roaming country changes
+    Then event notification "roaming-change-country" is received on callback-url
+    And sink credentials are received as expected
+    And notification body complies with the OAS schema at "##/components/schemas/RoamingChangeCountry"
+    And type="org.camaraproject.roaming-status-subscriptions.v0.roaming-change-country"
+
+@roaming_status_subscriptions_11_subscription_expiry
+  Scenario: Receive notification for subscription-ends event on expiry  
+    Given that subscriptions are created synchronously
+    And a valid subscription request body
+    And the request body property "$.subscriptionExpireTime" is set to a value in the near future
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    And the subscription is expired
+    And event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
+    And the response property "$.terminationReason" is "SUBSCRIPTION_EXPIRED"
+
+@roaming_status_subscriptions_12_subscription_ends_when_max_events_reached
+   Scenario: Receive notification for subscription-ends event on max events reached 
+    Given that subscriptions are created synchronously
+    And a valid subscription request body  
+    And the request body property "$.type" is "roaming_on"
+    And the request body property "$.subscriptionMaxEvents" is set to 1 
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    Then event notification "roaming_on" is received on callback-url
+    Then event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
+    And the response property "$.terminationReason" is "MAX_EVENTS_REACHED"
+
+  @roaming_status_subscriptions_13_subscription_delete_event_validation
+   Scenario: Receive notification for subscription-ends event on deletion 
+    Given that subscriptions are created synchronously
+    And a valid subscription request body  
+    When the request "createSubscription" is sent
+    Then the response code is 201 
+    When the request "deleteRoamingStatusSubscription" is sent
+    Then the response code is 202 or 204	
+    Then event notification "subscription-ends" is received on callback-url
+    And notification body complies with the OAS schema at "##/components/schemas/EventSubscriptionEnds"
+    And type="org.camaraproject.geofencing-subscriptions.v0.subscription-ends"
```

Comment:
The type should be `org.camaraproject.roaming-status-subscriptions.v0.subscription-ends`